.travis.yml

```diff
@@ -1,25 +1,64 @@
 language: java
 dist: trusty
 jdk:
-  - openjdk11
+- openjdk11
+env:
+- secure: J1qoSZK6VgeLDwanCtPTmV71GLyzkzGrF/NNj9F/YL+EKgnxCIW02WhYIrxVdiFBYx+KIcF7IlAwVnspMJ/m5+YtdgwK6yWwr0hfM6C7BOOsz4PKZNZbEogcv2P+ALmo6GHeRLo8XHXo52ML3sycsTTZC9CPcUC1PCADnRZvp5SVP+F9y9vOn7HfgmBPxhf5zTz0PiW2QGyD/tRaITCXkUtMcFqiu0N2xpBIDnuIn2X/ItXdLouPUMSP4JD3JLQBbrlW7zmNPK4RJMfyhEoiWBMwe6Q1s75Tk3g3DhaP5SZnlQAygMzMbWpPEmtMBvSQRIR0zaYRZAajvpcWGFgUhNFEj1KNQrzt/EVfB9oB80vnEiM7ux1OH3qZ0f+HWvsVO7iyz7F4QoCNUngmqv/KF9GfQduqHiXZ3piq5zNHC32xWDrNufu8h1PEuKKiVwEShRUni88JUdcoeujDI9ygp2v45gTXEQ13Nw2R5qVNWa8XM0vI/bvx0Y9PyiaL+LuxCyKn/k+as+rCSP1KCy2uJVC7SG6ZeYQKFeGDUdpYXqKonICGqGS/NKsZwPuY+917ydMs3KnU0Yiz2CZqDS9YdQuDCDGqj3gHNIGuF7QF6IOeNMMen1OyRqysqFPQsqDPd5T0/bLyGyYgAuyjeyoxp0w/CI2KRbNK4IAwMDHPxr4=
 before_cache:
-  - rm -f $HOME/.gradle/caches/modules-2/modules-2.lock
-  - rm -f $HOME/.gradle/caches/*/plugin-resoloution/
+- rm -f $HOME/.gradle/caches/modules-2/modules-2.lock
+- rm -f $HOME/.gradle/caches/*/plugin-resoloution/
 cache:
   directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper
+  - "$HOME/.gradle/caches/"
+  - "$HOME/.gradle/wrapper"
+before_install:
+# Download the install4j package needed for install building
+- wget --no-verbose https://download-gcdn.ej-technologies.com/install4j/install4j_unix_7_0_12.tar.gz
+#Put the bins where we can use them
+- tar -zxf install4j_unix_7_0_12.tar.gz
+#Make it findable later
+- export INSTALL4JDIR="$(pwd)/install4j7.0.12"
+#Download JRE's
+- wget --no-verbose "https://download.ej-technologies.com/bundles/java9/windows-amd64-11.0.3.tar.gz"
+- wget --no-verbose "https://download.ej-technologies.com/bundles/java9/macosx-amd64-11.0.3.tar.gz"
+- wget --no-verbose "https://otter-build-nightly.s3.amazonaws.com/linux-amd64-11.0.4.tar.gz"
 install:
   - git clone https://github.com/WycliffeAssociates/chromeabletabpane.git ../chromeabletabpane
   - git clone --branch=dev https://github.com/WycliffeAssociates/kotlin-resource-container.git ../kotlin-resource-container
   - git clone https://github.com/WycliffeAssociates/otter-common.git ../otter-common
   - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
   - echo "TRAVIS_BRANCH=$TRAVIS_BRANCH, PR=$PR, BRANCH=$BRANCH"
   - source init_common.sh
+# Grab the version from the version file for use later and concatenate the Travis build number to it
+  - if [ -f VERSION ]; then export VERSIONVAR="$(cat VERSION).$TRAVIS_BUILD_NUMBER"; fi
```

Comment:
a line feed does not affect it based on testing. Letters work. Symbols don't all work perfectly but they don't break the build. If install4j fails to build because it doesn't like what is passed in, then it will fail the build. If the upload to S3 doesn't like the filename the upload will fail, but this will not fail the build.

Reply:
Cool. Thanks for bringing it to my attention.